### ISSUE TITLE
Python binding fails if LD_LIBRARY_PATH is not present on Mac OS

### ISSUE THREAD

**Issue by xgfs** [dmlc/mxnet#1283]
I was installing mxnet on a clear mac system today, everything compiled but the python building were failing:
```python
  File "/Users/xgfs/apps/mxnet/example/image-classification/../../python/mxnet/libinfo.py", line 26, in find_lib_path
    elif os.name == "posix" and os.environ['LD_LIBRARY_PATH']:
  File "/Users/xgfs/anaconda/lib/python2.7/UserDict.py", line 23, in __getitem__
    raise KeyError(key)
KeyError: 'LD_LIBRARY_PATH'
```

Indeed, in the line 26 of libinfo.py environmental variable LD_LIBRARY_PATH is accessed before checked for validity. A simple fix would be
```python
elif os.name == "posix" and 'LD_LIBRARY_PATH' in os.environ and os.environ['LD_LIBRARY_PATH']:
```
adding the check before accessing the variable. After I did that, it worked like a charm. I have not found this problem referenced anywhere else, so I thought to submit it here.

**Comment by piiswrong** [comment]
@xgfs Thanks for reporting this. This is a recently introduce bug. Your fix looks good. Would you like to submit a PR?

**Comment by tqchen** [comment]
fixed by #1294
